packages/strapi-plugin-content-type-builder/admin/src/components/List/index.js

```diff
@@ -0,0 +1,164 @@
+/**
+*
+* List
+* params:
+*  -handlei18n: bool
+*   used for the buttonComponent to render label with FormattedMessage
+*  - listButtonLabel: string
+*  - listTitle: string
+*  - noListButtonPopUp: bool
+*     prevent from displaying the OldList button
+*  - renderRow: function
+*     overrides the default rendering of the OldList tr (we can pass customs components there)
+*  - listItems: array the elements to display
+*  - handleListPopButtonSave: func
+*
+*/
+
+import React from 'react';
+import { forEach, has, map} from 'lodash';
+import { FormattedMessage } from 'react-intl';
+
+import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
+import ButtonPrimaryHotline from 'components/Button';
+import PopUpForm from 'components/PopUpForm';
+import styles from './styles.scss';
+
+class List extends React.Component { // eslint-disable-line react/prefer-stateless-function
+  constructor(props) {
+    super(props);
+    this.state = {
+      modal: false,
+      isPopUpFormValid: true,
+      requiredInputs: [],
+    };
+  }
+
+  componentDidMount() {
+    const requiredInputs = [];
```

Comment:
Still not a big fan of the use of `forEach` loop to populate an array.

```js
const requiredInputs = Object.keys(this.props.sections).reduce((acc, current) => {
  acc.concat(
    this.props.sections[current].items
      .filter(item => has(item.validations, 'required'))
      .map(item => item.target)
  );
  
  return acc;
}, []);
```

Reply:
This code doesn't exist anymore. BTW it's a component from the settings-manager